index/secondary_index_manager.cc

```diff
@@ -341,4 +347,24 @@ bool secondary_index_manager::is_global_index(const schema& s) const {
     });
 }
 
+std::optional<sstring> secondary_index_manager::custom_index_class(const schema& s) const {
+
+    auto idx =_indices.find(index_name_from_table_name(s.cf_name()));
+
+    if (idx == _indices.end() || !(*idx).second.metadata().options().contains(cql3::statements::index_target::custom_index_option_name)) {
+        return std::nullopt;
+    } else {
+        return (*idx).second.metadata().options().at(cql3::statements::index_target::custom_index_option_name);
+    }
+}
+
+std::set<std::string_view> secondary_index_manager::supported_custom_classes(const gms::feature_service& fs) const {
+    using namespace std::literals;
+    const static std::set<std::string_view> classes = {
+        "usearch_vector_index"sv,
+        "opensearch_vector_index"sv,
```

Comment:
Maybe we should hide the implementation and just have vector_index here, and have the database configuration (db/config.cc) select the implementation.

Reply:
/cc @tzach

Reply:
What about this?